nni/mutable/shortcut.py

```diff
@@ -0,0 +1,235 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""High-level API for mutables."""
+
+from __future__ import annotations
+
+__all__ = [
+    'choice', 'randint', 'uniform', 'quniform', 'loguniform', 'qloguniform',
+    'normal', 'qnormal', 'lognormal', 'qlognormal'
+]
+
+import logging
+from typing import TYPE_CHECKING, TypeVar
+
+from .mutable import Categorical, Numerical
+
+if TYPE_CHECKING:
+    from nni.nas.nn.pytorch import LayerChoice
+
+T = TypeVar('T')
+
+_logger = logging.getLogger(__name__)
+
+
+def choice(label: str, choices: list[T]) -> Categorical[T] | LayerChoice:
+    """Choose from a list of options.
+
+    By default, it will create a :class:`~nni.mutable.Categorical` object.
+    ``choices`` should be a list of numbers or a list of strings.
+    Using arbitrary objects as members of this list (like sublists, a mixture of numbers and strings, or null values)
+    should work in most cases, but may trigger undefined behaviors.
+    If PyTorch modules are presented in the choices, it will create a :class:`~nni.nas.nn.pytorch.LayerChoice`.
+
+    For most search algorithms, choice are non-ordinal.
+    Even if the choices are numbers, they will still be treated as individual options,
+    and their numeric values will be neglected.
+
+    Nested choices (i.e., choice inside one of the options) is not currently supported by this API.
+
+    Examples
+    --------
+    >>> nni.choice('x', [1, 2, 3])
+    Categorical([1, 2, 3], label='x')
+    >>> nni.choice([nn.Conv2d(3, 3, 3), nn.Conv2d(3, 3, 5)], label='conv')
+    LayerChoice(
+        label='conv'
+        (0): Conv2d(3, 3, kernel_size=(3, 3), stride=(1, 1))
+        (1): Conv2d(3, 3, kernel_size=(5, 5), stride=(1, 1))
+    )
+    """
+    # Comment out before nas.nn is merged.
+    # try:
+    #     from torch.nn import Module
+    #     if all(isinstance(c, Module) for c in choices):
+    #         from nni.nas.nn.pytorch import LayerChoice
+    #         return LayerChoice(choices, label=auto_label(label))
+
+    #     from torch import Tensor
+    #     if any(isinstance(c, Tensor) for c in choices):
+    #         raise TypeError(
+    #             'Please do not use choice to choose from tensors. '
+    #             'If you are using this in forward, please use `InputChoice` explicitly in `__init__` instead.')
+    # except ImportError:
+    #     # In case PyTorch is not installed.
+    #     pass
+
+    return Categorical(choices, label=label)
+
+
+def randint(label: str, lower: int, upper: int) -> Categorical[int]:
+    """Choosing a random integer between lower (inclusive) and upper (exclusive).
+
+    Currently it is translated to a :func:`choice`.
```

Comment:
I want to make it default to quniform. This shortcut is not used in NAS I guess?
To achieve this the mutable need to aware that the result should be converted to int.

Reply:
It's not straightforward for `nni.randint().random()` to return a float, rather than an integer.

Also see quniform: `nni.quniform(low - 0.5, high + 0.5, 1).int()`. I think it's somewhat complicated and requires `simplify()` to get the variable.

Reply:
And it's currently never used in NAS.